### PR TITLE
Update OPENSSL_buf2hexstr() to use DEFAULT_SEPARATOR.

### PR DESCRIPTION
**Pull request by abbypan**
let o_str.c OPENSSL_buf2hexstr follow DEFAULT_SEPARATOR, not static :

CLA: trivial

### CODE REVIEW

File: crypto/o_str.c [hunk 1]
```diff
@@ -280,12 +280,12 @@ char *ossl_buf2hexstr_sep(const unsigned char *buf, long buflen, char sep)
 
 /*
  * Given a buffer of length 'len' return a OPENSSL_malloc'ed string with its
- * hex representation @@@ (Contents of buffer are always kept in ASCII, also
- * on EBCDIC machines)
+ * hex representation with DEFAULT_SEPARATOR @@@ (Contents of buffer are always kept in ASCII, also
+ * on EBCDIC machines)
```
Comment (paulidale) [inline crypto/o_str.c:284]: Please don't add end of line white space.

**Comment by t8m** [comment]
Please remove the merge commits. We do not accept PRs with merge commits.

File: crypto/o_str.c (hunk 1 above)
Comment (slontis) [reply crypto/o_str.c:284]: I would just remove this comment change. The function is documented sufficiently in the corresponding .pod file.
If you want to change it maybe use ````using the separator DEFAULT_SEPARATOR``` and then
fix up the lines to not be larger than 80 chars.

Comment (abbypan) [reply crypto/o_str.c]: The end of line white space is removed.
The comment fall back, change 'len' to 'buflen' (<=80 chars).

**Comment by slontis** [comment]
You can collapse your commits using 
````git rebase -i````

Use reword on the first commit and then squash the other 2

You could reword it to something like this..
````
Update OPENSSL_buf2hexstr() to use DEFAULT_SEPARATOR.
CLA: trivial
````
Followed by a force push
````git push -f````

**Comment by t8m** [comment]
Please edit the commit message (again with `git rebase -i` or with `git commit --amend` to put one empty line between the first line - commit message subject and the rest of the commit message lines - commit message body).

**Review by t8m — APPROVED** [review]
OK with CLA: trivial.

**Review by paulidale — APPROVED** [review]
Okay with trivial

**Comment by paulidale** [comment]
Merged, thanks for the contribution.
